### PR TITLE
 SPARK-30434 Move pandas related functionalities into 'pandas' sub-package

### PR DESCRIPTION
**Pull request by zero323**
@HyukjinKwon If it is not to much to ask, I'd really appreciate if you could take a quick glance and let me know if you see any obvious issues.

### CODE REVIEW

File: doc/api-coverage.rst [hunk 1]
```diff
@@ -150,6 +150,22 @@ API Coverage
 +------------------------------------------------+---------------------+--------------------+------------+
 | `pyspark.sql.group`_                           | ✘                   | ✔                  |            |
 +------------------------------------------------+---------------------+--------------------+------------+
+| `pyspark.sql.pandas`_                          | ✔                   | ✘                  |            |
```
Comment (HyukjinKwon) [inline doc/api-coverage.rst]: I intended `pandas` to be `Internal` too; however, I am maybe ignorant about the meaning of `Internal` and `Mixed`.

File: test-data/unit/sql-udf.test [hunk 2]
```diff
@@ -75,7 +75,7 @@ pandas_udf(lambda *xs: 42, "str", PandasUDFType.GROUPED_AGG)
 
 
 [case mapIterUdf]
-from pyspark.sql.functions import pandas_udf, PandasUDFType
```
Comment (HyukjinKwon) [inline test-data/unit/sql-udf.test]: The original import should be able to use as is without any change.

File: test-data/unit/sql-udf.test [hunk 3]
```diff
@@ -75,7 +75,7 @@ pandas_udf(lambda *xs: 42, "str", PandasUDFType.GROUPED_AGG)
...
+from pyspark.sql.pandas.functions import pandas_udf, PandasUDFType
```
Comment (HyukjinKwon) [inline test-data/unit/sql-udf.test]: I am also not so much aware of this code base; however, just wanted to note:

- original import will work as was.
- MAP ITER became `mapInPandas`

Comment (zero323) [reply test-data/unit/sql-udf.test]: > original import will work as was.

That's primarily to check both sides. In general MyPy is pretty strict about these things so

```
from pyspark.sql.pandas.functions import pandas_udf, PandasUDFType
```
would type check fine in `pyspark.sql.functions` but, not as transitive import. For that I had to switch to aliases.
```
from pyspark.sql.pandas.functions import pandas_udf as pandas_udf, PandasUDFType as PandasUDFType
```

**Comment by zero323** [comment]
> Looks good from a cursory look.

I couldn't ask for more. Thank you so much @HyukjinKwon!

**Pull request merged**